### PR TITLE
[DOCS] Fixes small friction points noticed during onboarding

### PR DESCRIPTION
**Pull request by klavavej**
This issueless PR addresses a variety of small friction points I encountered during onboarding. The changes mostly fall into a few themes:

- improving relevance of links
- fixing indentation around lists
- spelling/grammar fixes (a couple code syntax fixes as well)
- for code samples, making sure inline code in surrounding prose commentary matches the contents of the associated code block verbatim (there were several slightly mismatched variable names and mismatches of single vs double quotes)
- removing language that tells people things are easy (It's better to show that something is easy by presenting it in clear concise steps than to explicitly tell people something is easy. Explicitly saying something is easy could frustrate folks who struggle to follow the instructions for any reason.)
- wording tweaks for clarity/accuracy
- adding language identifiers to code blocks for consistency 

------

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: docs/docusaurus/docs/reference/learn/integrations/dbt_tutorial.md [hunk 1]
```diff
@@ -378,7 +378,7 @@ Add Postgres credentials:
     - Connection id: postgres
     - Connection Type: Postgres
     - Host: database
-    - Schema: postgres
+    - Database: postgres
```
Comment (klavavej) [inline docs/docusaurus/docs/reference/learn/integrations/dbt_tutorial.md:381]: Note for reviewer: This appears to have changed in the airflow UI. Here's a side by side of airflow vs the screenshot in the docs. (I don't think the mismatch is severe enough to warrant the effort to update the screenshot, but I do think it's worth updating the prose)
Screenshot 2024-10-24 at 9 09 30 AM

File: docs/docusaurus/docs/cloud/validations/manage_validations.md [hunk 2]
```diff
@@ -79,6 +79,3 @@ To make the identification of Data Assets containing partitions easier, partitio
 6. Optional. Hover over a circle in the Validation timeline to view details about a specific Validation run, including the observed values.
 
     ![Validation timeline detail](/img/view_validation_timeline_detail.png)
-
-7. Optional. To hide the Validation timeline, click the **Validation timeline** (![Validation timeline icon](/img/validation_timeline.png)) icon.
```
Comment (klavavej) [inline docs/docusaurus/docs/cloud/validations/manage_validations.md:83]: note for reviewer: I removed this optional step since this icon no longer appears in the UI 
Screenshot 2024-10-24 at 9 12 56 AM

File: docs/README.md [hunk 3]
```diff
@@ -2,9 +2,9 @@
 
 If you’ve noticed an issue in the documentation, or have recommendations for improving the content, you can submit your request as a GitHub issue in this repository. To avoid forking the repository, Great Expectations recommends using the GitHub Markdown editor to edit documentation.
 
-1. Go to the topic file in the [GitHub Great Expectations docs repository](https://github.com/great-expectations/great_expectations/tree/develop/docs) you want to edit. 
+1. In the [`docusaurus/docs` folder](https://github.com/great-expectations/great_expectations/tree/develop/docs), go to the topic file you want to edit. 
 
-    The topic URL includes the topic file name. For example, the URL for the Great Expectations Quickstart is https://docs.greatexpectations.io/docs/tutorials/quickstart. The URL indicates the topic is located in the `tutorials` folder, and it's named `quickstart`.
+    The topic URL includes the topic file name. For example, the URL for the Great Core overview is https://docs.greatexpectations.io/docs/core/introduction/gx_overview. The URL indicates the topic is located in the `core/introduction` folder, and it's named `gx_overview`.
```
Comment (klavavej) [inline docs/README.md]: Note for reviewer: I changed this because the previous example URL was redirecting which made this example a bit confusing.

File: docs/docusaurus/static/docs/oss/images/conditional_data_docs_screenshot.png
Comment (klavavej) [inline docs/docusaurus/static/docs/oss/images/conditional_data_docs_screenshot.png:1]: Note for reviewer: I was quite surprised to see "sex" in this screenshot. That part of the image is not integral to what this screenshot is meant to illustrate so I cropped it out to reduce risk that readers are shocked/distracted

File: docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_data_source/_datasource_method_reference_table.md [hunk 4]
```diff
@@ -2,6 +2,6 @@
 |------------------|------------------------------------------------------------------------------|
 | PostgreSQL       | `context.data_sources.add_postgres(name: str, connection_string: str)`       |
 | SQLite | `context.data_sources.add_sqlite(name: str, connection_string: str)`         |
-| DataBricks SQL   | `context.data_sources.add_databricks_sql(name: str, connection_string: str)` |
 | Snowflake        | `context.data_sources.add_snowflake(name:str, connection_string: str)`       |
+| DataBricks SQL   | `context.data_sources.add_databricks_sql(name: str, connection_string: str)` |
```
Comment (klavavej) [inline docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_data_source/_datasource_method_reference_table.md:6]: note for reviewer: I moved this so the order of databases is consistent across both tables on this page

File: docs/docusaurus/docs/core/define_expectations/_examples/retrieve_a_batch_of_test_data_from_a_batch_definition.py [hunk 5]
```diff
@@ -71,6 +71,6 @@ def set_up_context_for_example(context):
 # </snippet>
 
 # <snippet name="docs/docusaurus/docs/core/define_expectations/_examples/retrieve_a_batch_of_test_data_from_a_batch_definition.py - verify populated Batch">
-batch.head()
+print(batch.head())
```
Comment (klavavej) [inline docs/docusaurus/docs/core/define_expectations/_examples/retrieve_a_batch_of_test_data_from_a_batch_definition.py:74]: Note for reviewer: I added `print()` so that this will do what the introductory prose promises ("printing the header and first few records of the returned Batch"). This addresses the user feedback reported in https://greatexpectations.atlassian.net/jira/software/c/projects/DOC/issues/DOC-848

File: docs/docusaurus/docs/core/run_validations/run_a_validation_definition.md [hunk 6]
```diff
@@ -45,7 +45,7 @@ import PrereqValidationDefinition from '../_core_components/prerequisites/_valid
 
    Validation Results are automatically saved in your Data Context when a Validation Definition's `run()` method is called.  For convenience, the `run()` method also returns the Validation Results as an object you can review.
 
-   :::tip
+   :::tip Result verbosity
```
Comment (klavavej) [inline docs/docusaurus/docs/core/run_validations/run_a_validation_definition.md:48]: note for reviewer: I added more specific titles for these tips since there are two tips so close to each other on this page.

File: docs/docusaurus/docs/core/set_up_a_gx_environment/install_additional_dependencies.md [hunk 7]
```diff
@@ -13,7 +13,7 @@ import InstallSql from './_install_additional_dependencies/_sql.md'
 import InstallSpark from './_install_additional_dependencies/_spark.md'
 
 Some environments and Data Sources require additional Python libraries or third party utilities that are not included in the base installation of
-GX Core. Use the information provided here to install the necessary dependencies for Amazon S3, Azure Blob Storage, Google Cloud Storage, and SQL databases.
+GX Core. Use the information provided here to install the necessary dependencies for your databases.
```
Comment (klavavej) [inline docs/docusaurus/docs/core/set_up_a_gx_environment/install_additional_dependencies.md:16]: Note for reviewer: this list was missing one option that's present in the tabs that follow. To reduce maintenance burden for the future I made the intro line more generic so it won't have to be updated any time a new tab is added.

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -6,6 +6,6 @@
 - [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
 - [ ] Appropriate tests and docs have been updated
 
-For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).
+For more information about contributing, visit our [community resources](/docs/core/introduction/community_resources#contribute-code-or-documentation).
```
Comment (klavavej) [inline .github/pull_request_template.md:9]:
```suggestion
For more information about contributing, visit our [community resources](https://docs.greatexpectations.io/docs/core/introduction/community_resources#contribute-code-or-documentation).
```

File: docs/README.md [hunk 9]
```diff
@@ -2,9 +2,9 @@
 
 If you’ve noticed an issue in the documentation, or have recommendations for improving the content, you can submit your request as a GitHub issue in this repository. To avoid forking the repository, Great Expectations recommends using the GitHub Markdown editor to edit documentation.
 
-1. Go to the topic file in the [GitHub Great Expectations docs repository](https://github.com/great-expectations/great_expectations/tree/develop/docs) you want to edit. 
+1. In the [`docusaurus/docs` folder](https://github.com/great-expectations/great_expectations/tree/develop/docs), go to the topic file you want to edit.
```
Comment (klavavej) [inline docs/README.md:5]:
```suggestion
1. In the [`docusaurus/docs` folder](https://github.com/great-expectations/great_expectations/tree/develop/docs/docusaurus/docs), go to the topic file you want to edit. 
```

File: docs/README.md [hunk 10]
```diff
@@ -2,9 +2,9 @@
 
 If you’ve noticed an issue in the documentation, or have recommendations for improving the content, you can submit your request as a GitHub issue in this repository. To avoid forking the repository, Great Expectations recommends using the GitHub Markdown editor to edit documentation.
 
-1. Go to the topic file in the [GitHub Great Expectations docs repository](https://github.com/great-expectations/great_expectations/tree/develop/docs) you want to edit. 
+1. In the [`docusaurus/docs` folder](https://github.com/great-expectations/great_expectations/tree/develop/docs/docusaurus/docs), go to the topic file you want to edit. 
 
-    The topic URL includes the topic file name. For example, the URL for the Great Expectations Quickstart is https://docs.greatexpectations.io/docs/tutorials/quickstart. The URL indicates the topic is located in the `tutorials` folder, and it's named `quickstart`.
+    The topic URL includes the topic file name. For example, the URL for the Great Core overview is https://docs.greatexpectations.io/docs/core/introduction/gx_overview. The URL indicates the topic is located in the `core/introduction` folder, and it's named `gx_overview`.
```
Comment (klavavej) [inline docs/README.md:7]:
```suggestion
    The topic URL includes the topic file name. For example, the URL for the Great Expectations Core overview is https://docs.greatexpectations.io/docs/core/introduction/gx_overview. The URL indicates the topic is located in the `core/introduction` folder, and it's named `gx_overview`.
```

File: great_expectations/expectations/core/expect_table_column_count_to_be_between.py [hunk 11]
```diff
@@ -110,7 +110,7 @@ class ExpectTableColumnCountToBeBetween(BatchExpectation):
         Passing Case:
             Input:
                 ExpectTableColumnCountToBeBetween(
-                    min_value=1
+                    min_value=1,
```
Comment (klavavej) [inline great_expectations/expectations/core/expect_table_column_count_to_be_between.py:113]: **Question for reviewer**: I'm not sure where to preview this change nor am I confident it will have the desired effect. The expectations gallery seems to be outside the scope of the deploy preview for this PR https://deploy-preview-10560.docs.greatexpectations.io/

**Is this the right way to make changes to expectation gallery entries?**

File: docs/docusaurus/docs/core/connect_to_data/filesystem_data/_create_a_batch_definition/_examples/_directory_partitioned_daily.py [hunk 12]
```diff
@@ -20,6 +20,6 @@
 batch = batch_definition.get_batch(
     batch_parameters={"year": "2019", "month": "01", "day": "01"}
 )
-batch.head()
+print(batch.head())
```
Comment (klavavej) [inline docs/docusaurus/docs/core/connect_to_data/filesystem_data/_create_a_batch_definition/_examples/_directory_partitioned_daily.py:23]: **Question for reviewer:** these changes to `batch.head()` aren't showing where I'd expect them to, for example on https://deploy-preview-10560.docs.greatexpectations.io/docs/core/connect_to_data/filesystem_data/ 

**Am I missing some step to make the change take effect or am I looking for it in the wrong place?**

File: docs/docusaurus/docs/core/configure_project_settings/access_secrets_managers/_gcp_secret_manager.md [hunk 13]
```diff
@@ -9,9 +9,9 @@ import PreReqFileDataContext from '../../_core_components/prerequisites/_file_da
 
 ### Procedure
 
-1. Set up Azure Key Vault support.
```
Comment (tyler-hoffman) [inline docs/docusaurus/docs/core/configure_project_settings/access_secrets_managers/_gcp_secret_manager.md:12]: Oh yikes! Thanks especially for this change!

File: docs/docusaurus/docs/core/connect_to_data/filesystem_data/_create_a_batch_definition/_examples/_directory_partitioned_daily.py (hunk 12 above)
Comment (tyler-hoffman) [reply docs/docusaurus/docs/core/connect_to_data/filesystem_data/_create_a_batch_definition/_examples/_directory_partitioned_daily.py:23]: What *are* you seeing?

Comment (tyler-hoffman) [reply docs/docusaurus/docs/core/connect_to_data/filesystem_data/_create_a_batch_definition/_examples/_directory_partitioned_daily.py:23]: fwiw I don't typically use `head()`, but I just tried, and what I'm seeing is that looking at the value (e.g. in a debugger, you'll see reasonable data, but printing is doing nothing. Seems like a bug to me that print does nothing - may worth filing a ticket on Timber's board. I'd probably leave off the `print` for now.

Comment (tyler-hoffman) [reply docs/docusaurus/docs/core/connect_to_data/filesystem_data/_create_a_batch_definition/_examples/_directory_partitioned_daily.py:23]: oops, I take it back - I didn't see anything printed because of something about my environment. This should work. Here's my sample code:

```
batch = (
    context.data_sources.add_pandas(name="my_datasource")
    .add_dataframe_asset(name="my_asset")
    .add_batch_definition_whole_dataframe(name="my_batch")
    .get_batch(batch_parameters={"dataframe": pd.DataFrame({"a": [1, 2]})})
)

print("==")
print(batch.head())
print("==")
```

and the output (the call to head itself prints the stuff about calculating metrics):

```
==
Calculating Metrics: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 257.94it/s]
   a
0  1
1  2
==
```

File: great_expectations/expectations/core/expect_table_column_count_to_be_between.py (hunk 11 above)
Comment (tyler-hoffman) [reply great_expectations/expectations/core/expect_table_column_count_to_be_between.py:113]: @JessSaavedra I bet you have an answer on this

**Review by tyler-hoffman — APPROVED** [review]
Looks great to me. Thanks!

File: great_expectations/expectations/core/expect_table_column_count_to_be_between.py (hunk 11 above)
Comment (JessSaavedra) [reply great_expectations/expectations/core/expect_table_column_count_to_be_between.py:113]: right now we don't have an automatized process to impact these changes in the expectations gallery, we run a script manually to regenerate the `expectations-list.js` file in the expectations gallery from these files (so yes, it's ok to make the changes here). After you've merged this PR we can do the update so that it gets reflected in the gallery :smile:

File: docs/docusaurus/docs/core/connect_to_data/filesystem_data/_create_a_batch_definition/_examples/_directory_partitioned_daily.py (hunk 12 above)
Comment (klavavej) [reply docs/docusaurus/docs/core/connect_to_data/filesystem_data/_create_a_batch_definition/_examples/_directory_partitioned_daily.py:23]: > What are you seeing?

I am seeing `batch.head()` still in the deploy preview code samples. I'm not seeing the change I'm trying to apply to wrap that in `print()`.

It looks like there are at least 4 layers of files involved in the creation of the page where I'm expecting to see changes so I may just be getting my wires crossed but here's a screenshot of one case where I'm not seeing  the expected change
Screenshot 2024-10-25 at 2 14 18 PM

From left to right in vs code I have
- the main markdown file for the page at https://deploy-preview-10560.docs.greatexpectations.io/docs/core/connect_to_data/filesystem_data/
- a markdown snippet for the "Create a Batch Definition" section of the page
- a markdown snippet for the "Directory data asset" tab of that section
- a .py file for the code samples that display in the Partioned > Daily subtab of the section. 

Then at the bottom I have what I'm seeing in the deploy preview

---

Am I looking in the wrong place in the deploy preview to see this change take effect? Do I need to do something additional to make the change take effect?

Comment (klavavej) [reply docs/docusaurus/docs/core/connect_to_data/filesystem_data/_create_a_batch_definition/_examples/_directory_partitioned_daily.py:23]: after talking to @rachhouse I cleared the netlify cache and retried the build and that fixed things. I now see this change where I expect to see it

File: great_expectations/expectations/core/expect_table_column_count_to_be_between.py (hunk 11 above)
Comment (klavavej) [reply great_expectations/expectations/core/expect_table_column_count_to_be_between.py:113]: Thanks @JessSaavedra ! I just merged this PR. Would you please run the script to update the gallery?
